### PR TITLE
Support revoked proofs

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

When a proof's revoked, it doesn't get sent to us in the proofs array; it's not a real proof anymore, that kind of info isn't available for it.  But now (since CORE-2760) we get a revokedProofs array with some of the same info in.  So the task of this PR is to take revokedProofs and synthesize Proofs out of its members, so that we can show a revoked proof inside the popup.  Previously to this PR we didn't show revoked proofs at all, and it looked like you were being asked to track someone you didn't already track.

Doing this uncovered two more bugs -- DESKTOP-756 and DESKTOP-757, the first of which should be fixed before merging this one in.

Before:
screenshot from 2016-03-18 15-15-24

After:
screenshot from 2016-03-18 15-15-50

### CODE REVIEW

File: shared/reducers/tracker.js [hunk 1]
```diff
@@ -446,11 +463,11 @@ function remoteProofToProof (rp: RemoteProof, lcr: ?LinkCheckResult): Proof {
     state: proofState,
     id: rp.sigID,
     meta: statusMeta || diffMeta,
-    type: proofType,
+    type: remoteProofToProofType(rp),
     color: stateToColor(proofState),
     name: rp.displayMarkup,
     humanUrl: humanUrl,
-    profileUrl: rp.displayMarkup && proofUrlToProfileUrl(rp.proofType, rp.displayMarkup, rp.key, humanUrl)
+    profileUrl: rp.displayMarkup && proofUrlToProfileUrl(rp.proofType.toString(), rp.displayMarkup, rp.key, humanUrl)
```
Comment (MarcoPolo) [inline shared/reducers/tracker.js]: Why did you have to add `.toString()`?

Comment (cjb) [inline shared/reducers/tracker.js]: @MarcoPolo Here's the flow error if I don't:

```
 λ flow

reducers/tracker.js:470
470:     profileUrl: rp.displayMarkup && proofUrlToProfileUrl(rp.proofType, rp.displayMarkup, rp.key, humanUrl)
                                                              ^^^^^^^^^^^^ number literal `0`. This type is incompatible with
412: function proofUrlToProfileUrl (proofType: string, name: string, key: ?string, humanUrl: ?string): string {
                                               ^^^^^^ string

[.. more errors for the other numbers in the enum]

reducers/tracker.js:470
470:     profileUrl: rp.displayMarkup && proofUrlToProfileUrl(rp.proofType, rp.displayMarkup, rp.key, humanUrl)
                                                              ^^^^^^^^^^^^ number literal `6`. This type is incompatible with
412: function proofUrlToProfileUrl (proofType: string, name: string, key: ?string, humanUrl: ?string): string {
                                               ^^^^^^ string


Found 11 errors
```

But I don't see why my change to create remoteProofToProofType() provoked the error.  Any ideas, something else I should try?

**Comment by MarcoPolo** [comment]
:+1: % Change proofUrlToProfileUrl's first argument type to be a `number` instead of `string` and don't use `.toString()`

**Comment by cjb** [comment]
@MarcoPolo thanks, fixed proofUrlToProfileUrl!

**Pull request merged**